### PR TITLE
Moc openstack login  -   Depends on tox/django test

### PR DESCRIPTION
**Pull request by rob-baron**
I have moved the changes for the open stack login here.  The unit tests work on my system, need to adjust travis ci file.

### CODE REVIEW

File: api/test/views/mock_keystone.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        if token == 'gAAAAABY2WAKlR6yxvn1mvkPcW0i4fB36y68N0mg2u1CUjPg7OJ-alV2CPPg207_zOTm8s3p3hHJWYVCw5Jh3PBYECkdplVgdzH7UgviSftNM8AJboFNXbypNNaELqgSztpGh5TnBbXI5RlzAc4IIBWd8wPL-RZuOA'\
+                and user_id == "6705630a653f4300baf486a8df6072de":
+            json_text = '{"links": {"self": "http://localhost:8082/api/test/keystone/v3/users/6705630a653f4300baf486a8df6072de/projects", "previous": null, "next": null}, "projects":'\
+                        ' [{"is_domain": false, "description": "a test", "links": {"self": "http://localhost:8082/api/test/kestone/v3/projects/190ce9f5a454493e9eaae608d54fe2d1"},'\
+                        ' "enabled": true, "id": "190ce9f5a454493e9eaae608d54fe2d1", "parent_id": "default", "domain_id": "default", "name": "MockeyMock"}]}'
+            response = HttpResponse(json_text, status=status.HTTP_200_OK, content_type="application/json")
+            response['x-openstack-request-id'] = "req-6a536bef-af21-46a9-94e3-31f47998644f"
+        elif token == 'gAAAAABY2WAKlR6yxvn1mvkPcW0i4fB36y68N0mg2u1CUjPg7OJ-alV2CPPg207_zOTm8s3p3hHJWYVCw5Jh3PBYECkdplVgdzH7UgviSftNM8AJboFNXbypNNaELqgSztpGh5TnBbXI5RlzAc4IIBWd8wPL-RZuOA':
```
Comment (lucashxu) [inline api/test/views/mock_keystone.py]: I am bit confused with this part. Here is checking whether the token is equal to a uuid. A uuid cannot be reproduced. so how this check gonna work?

File: api/test/views/mock_keystone.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+    def post(self, request):
+        logger.info("MockKeystoneView:post called")
+        logger.info(request.body)
+        return JsonResponse({'text': 'get response'})
+
+
+class MockKeystoneUsersProjects(APIView):
+    """
+    This is just a simple view to catch any url we are not handling
```
Comment (knikolla) [inline api/test/views/mock_keystone.py]: update docstring to mention the url your are mocking and what it's doing.

**Comment by xuhang57** [comment]
Are we doing this:
https://github.com/openstack/django_openstack_auth/blob/master/openstack_auth/tests/tests.py

Or doing some similar thing here for testing?

**Comment by xuhang57** [comment]
https://github.com/openstack/django_openstack_auth/blob/master/openstack_auth/tests/data_v3.py 

If you could have a similar file like this

File: api/test/views/mock_keystone.py (hunk 2 above)
Comment (lucashxu) [reply api/test/views/mock_keystone.py]: @knikolla  Should we use port 5000 rather than 8082? Or it doesn't really matter?

**Comment by xuhang57** [comment]
And why we want to put this test file into `GUI-Backend` but not into the django_cyverse_auth`?  Or even we have our `django_moc_auth` just like OpenStack has `django_openstack_auth`?

File: api/test/views/mock_keystone.py (hunk 2 above)
Comment (knikolla) [reply api/test/views/mock_keystone.py]: @xuhang57 doesn't matter.

File: api/test/views/mock_keystone.py (hunk 1 above)
Comment (knikolla) [reply api/test/views/mock_keystone.py]: @xuhang57 This is not a uuid. But anyway, this is the token we are returning from mock keystone, therefore it IS the token that atmosphere will use for all communications.

**Comment by rob-baron** [comment]
We could also just use their test suite, sure.  However, this is how I tested the entire atmosphere + django-cyverse-auth backend.

**Comment by rob-baron** [comment]
There are 2 sets of files.  I have placed the MocLogin test file in the atmosphere/tests directory as it is testing the the atmosphere web api.  Granted Django-Cyvers-Auth accesses the keystone client, but so does atmosphere.  I planned to use this to test the projects as well, which are accessing keystone auth from celery.  It makes no sense to place this solely in django-cyverse-auth.

I have placed the mock keystone in api/test for the following reasons:
    1) This is already a rest api.
    2) I didn't want to rely on an external web server, or anything that required more configuration effort.
    3) It can be conditionally included based off of the configuration file.
    4) it is interacting with the keystone client and not strictly django-cyverse-auth.

File: tests/test_MocLogin.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            virtualization=kvm,
+            type=openstack_type,
+            public=True)[0]
+        MOC.cloud_config = \
+            '{"user": {"domain": "default", "secret": "012345678901234567890123456789012",'\
+            ' "user_role_name": "_member_", "admin_role_name": "admin"},'\
+            ' "deploy": {"hostname_format": "%(one)s.%(two)s.%(three)s.%(four)s"},'\
+            ' "network": {"topology": "External Network Topology",'\
+            '     "dns_nameservers": ["8.8.8.8", "8.8.4.4"],'\
+            '     "default_security_rules": [["ICMP", -1, -1], ["UDP", 20, 20], ["TCP", 20, 21], ["TCP", 22, 23],'\
```
Comment (lucashxu) [inline tests/test_MocLogin.py:37]: we don't need these default security rules

**Review by xuhang57 — CHANGES_REQUESTED** [review]

**Comment by xuhang57** [comment]
a plus one after taking out the security group rules that we don't need

**Comment by xuhang57** [comment]
resolve conflict and a plus one. feel free to merge after that.

File: tests/test_MocLogin.py (hunk 3 above)
Comment (rob-baron) [reply tests/test_MocLogin.py:37]: OK, I have clipped them to a subset.  Should have a few so that we accurately represent what the cloud_config contains.

**Review by xuhang57 — APPROVED** [review]

**Review by knikolla — APPROVED** [review]

**Pull request merged**